autotest/ogr/ogr_sql_test.py

```diff
@@ -130,14 +130,13 @@ def test_ogr_sql_1():
     gdaltest.ds = ogr.Open("data")
     gdaltest.lyr = gdaltest.ds.GetLayerByName("poly")
 
-    gdaltest.lyr.SetAttributeFilter("eas_id < 167")
+    with gdaltest.lyr.SetAttributeFilter("eas_id < 167"):
 
-    count = gdaltest.lyr.GetFeatureCount()
-    assert count == 3, (
-        "Got wrong count with GetFeatureCount() - %d, expecting 3" % count
-    )
+        count = gdaltest.lyr.GetFeatureCount()
+        assert count == 3, (
+            "Got wrong count with GetFeatureCount() - %d, expecting 3" % count
+        )
 
-    gdaltest.lyr.SetAttributeFilter("")
```

Comment:
@dbaston if test_ogr_sql_1 were split into two tests, instead of combining two cases into one, reset of the layer's attribute filter would be a non-issue.